lib/HTFeed/Volume.pm

```diff
@@ -28,748 +28,825 @@ my $stage_map = undef;
+	'ingest_date',
+	'jhove_files',
+	'page_data',
+	'repository_symlink',
+	'repsitory_mets_xpc',
+	'source_mets_file',
+	'source_mets_xpc',
+	'utf8_files',
+    );
+    map { delete $self->{$_} } @fields;
 }
 
 sub get_identifier {
-  my $self = shift;
-  return $self->get_namespace() . q{.} . $self->get_objid();
-
+    my $self = shift;
+    join(".", $self->get_namespace(), $self->get_objid());
 }
 
 sub get_namespace {
-  my $self = shift;
-  return $self->{namespace};
+    my $self = shift;
+    return $self->{namespace};
 }
 
 sub get_objid {
-  my $self = shift;
-  return $self->{objid};
+    my $self = shift;
+    return $self->{objid};
 }
 
 sub get_packagetype {
-  my $self = shift;
-  return $self->{packagetype};
+    my $self = shift;
+    return $self->{packagetype};
 }
 
 sub get_pt_objid {
-  my $self = shift;
-  return s2ppchars($self->{objid});
+    my $self = shift;
+    return s2ppchars($self->{objid});
 }
 
 sub get_file_groups {
-  my $self = shift;
-
-  if(not defined $self->{filegroups}) {
-    my $filegroups = {}; 
-
-    my $nspkg_filegroups = $self->{nspkg}->get('filegroups');
-    while( my ($key,$val) = each (%{ $nspkg_filegroups })) {
-      my $files = [];
-      my $re = $val->{file_pattern} or die("Missing pattern for filegroup $key");
-      foreach my $file ( @{ $self->get_all_directory_files() }) {
-        push(@$files,$file) if $file =~ $re;
-      }
-      $filegroups->{$key} = new HTFeed::FileGroup($files,%$val);
+    my $self = shift;
+
+    if(not defined $self->{filegroups}) {
+	my $filegroups = {};
+
+	my $nspkg_filegroups = $self->{nspkg}->get('filegroups');
+	while( my ($key, $val) = each (%{ $nspkg_filegroups })) {
+	    my $files = [];
+	    my $re = $val->{file_pattern} or die("Missing pattern for filegroup $key");
+	    foreach my $file ( @{ $self->get_all_directory_files() }) {
+		push(@$files, $file) if $file =~ $re;
+	    }
+	    $filegroups->{$key} = new HTFeed::FileGroup($files, %$val);
+	}
+	$self->{filegroups} = $filegroups;
     }
-    $self->{filegroups} = $filegroups;
-  }
 
-  return $self->{filegroups};
+    return $self->{filegroups};
 }
 
 sub get_all_directory_files {
-  my $self = shift;
-  my $path = shift;
-
-#    if(not defined $self->{directory_files}) {
-  $self->{directory_files} = [];
-  my $stagedir = $path || $self->get_staging_directory();
-  opendir(my $dh,$stagedir) or croak("Can't opendir $stagedir: $!");
-  foreach my $file (readdir $dh) {
-    # ignore ., .., Mac .DS_Store files
-    push(@{ $self->{directory_files} },$file) unless ($file =~ /^\.+$/ or $file eq '.DS_Store' or $file =~ /^thumbs\.db$/i);
-  }
-  closedir($dh) or croak("Can't closedir $stagedir: $!");
-  @{ $self->{directory_files} } = sort( @{ $self->{directory_files} } );
-#    }
+    my $self = shift;
+    my $path = shift;
+
+    $self->{directory_files} = [];
+    my $stagedir = $path || $self->get_staging_directory();
+    opendir(my $dh,$stagedir) or croak("Can't opendir $stagedir: $!");
```

Comment:
This seems to be an exception to the "space after param comma" standard, which I also endorse. Ditto my comment about searching for other exceptions, there seem to be some but not a lot.